app/middleware/headerAndFooter.js

```diff
@@ -0,0 +1,56 @@
+const rp = require('request-promise-native');
+const NodeCache = require('node-cache');
+
+const { endpoint, cacheTimeout, subscriptionKey } = require('../../config/config').headerFooterApi;
+
+const headerFooterCache = new NodeCache();
+
+module.exports = async (_, res, next) => {
+  try {
+    // Check for cached data
+    const cachedData = headerFooterCache.get('nhsHeaderFooterApi');
+    if (cachedData) {
+      // Attach cached data to response
+      res.locals = {
+        ...res.locals,
+        footerLinks: cachedData.footerLinks,
+        headerLinks: cachedData.headerLinks,
+      };
+      return next();
+    }
+
+    // Get data from Wagtail endpoint
+    const { header, footer } = await rp(endpoint, {
+      headers: {
+        'Content-Type': 'application/json',
+        'subscription-key': subscriptionKey,
```

Comment:
The key for headers (I believe is case sensitive) so it should be Subscription-Key (capital S)

Reply:
Ahh we use lowercase on all our API documentation but I will change it to be consistent with Nuget package 👍